api/clients/v2/examples/example_relay_retrieval_test.go

```diff
@@ -0,0 +1,86 @@
+package examples
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/status-im/keycard-go/hexutils"
+)
+
+// This example demonstrates how to use the RelayPayloadRetriever to retrieve a payload from EigenDA, running on
+// holesky testnet
+func Example_relayPayloadRetrieval() {
+	// You must provide a private key that either has a testnet reservation, or you must configure on-demand payments
+	// by sending funds to the payment vault.
+	privateKey := ""
+
+	// Create a payload disperser and disperse a sample payload to EigenDA
+	// This will be the payload we will later retrieve
+	payloadDisperser, err := createPayloadDisperser(privateKey)
+	if err != nil {
+		panic(fmt.Sprintf("create payload disperser: %v", err))
+	}
+	defer payloadDisperser.Close()
+
+	payload, err := createRandomPayload(4 * 1024) // (4KB of random data)
+	if err != nil {
+		panic(fmt.Sprintf("create random payload: %v", err))
+	}
+
+	dispersalCtx, dispersalCancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer dispersalCancel()
+	eigenDACert, err := payloadDisperser.SendPayload(dispersalCtx, payload)
+	if err != nil {
+		panic(fmt.Sprintf("send payload: %v", err))
+	}
+
+	fmt.Printf("Successfully dispersed payload\n")
+
+	// Create a payload retriever to retrieve from EigenDA relays
+	payloadRetriever, err := createRelayPayloadRetriever()
+	if err != nil {
+		panic(fmt.Sprintf("create relay payload retriever: %v", err))
+	}
+	defer payloadRetriever.Close()
+
+	retrievalCtx, retrievalCancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer retrievalCancel()
+	// Retrieve the payload using the certificate
+	retrievedPayload, err := payloadRetriever.GetPayload(retrievalCtx, eigenDACert)
+	if err != nil {
+		panic(fmt.Sprintf("get payload: %v", err))
+	}
+
+	// Verify that the retrieved payload matches the original by comparing bytes
+	originalBytes := payload.Serialize()
+	retrievedBytes := retrievedPayload.Serialize()
+	if !bytes.Equal(originalBytes, retrievedBytes) {
+		panic(fmt.Sprintf(
+			"retrieved payload doesn't match original payload (original: %s, retrieved: %s)",
+			hexutils.BytesToHex(originalBytes), hexutils.BytesToHex(retrievedBytes)))
+	}
```

Comment:
is this code needed? what purpose does it serve?

Reply:
typically retrievers are different people who don't have access to the original payload, and would thus need to verify the commitment instead of doing this simple check here (which I don't think anyone in any scenario would ever do right?)

Reply:
Fair point. Removed byte comparison in [bb927c02](https://github.com/Layr-Labs/eigenda/pull/1454/commits/bb927c02ee982f804ded8f4f242e9aeedb9c7fe5)